embeddings/embedding.go

```diff
@@ -91,28 +91,12 @@ func BatchedEmbed(ctx context.Context, embedder EmbedderClient, texts []string,
 	batchedTexts := BatchTexts(texts, batchSize)
 
 	emb := make([][]float32, 0, len(texts))
-	for _, texts := range batchedTexts {
-		curTextEmbeddings, err := embedder.CreateEmbedding(ctx, texts)
+	for _, batch := range batchedTexts {
+		curBatchEmbeddings, err := embedder.CreateEmbedding(ctx, batch)
 		if err != nil {
 			return nil, err
 		}
-		// If the size of this batch is 1, don't average/combine the vectors.
-		if len(texts) == 1 {
-			emb = append(emb, curTextEmbeddings[0])
-			continue
-		}
-
-		textLengths := make([]int, 0, len(texts))
-		for _, text := range texts {
-			textLengths = append(textLengths, len(text))
-		}
-
-		combined, err := CombineVectors(curTextEmbeddings, textLengths)
```

Comment:
This commit entirely removes the `CombineVectors` functionality that was here before.

In fact, `CombineVectors` now becomes unused; why is this?

Reply:
@tmc FYI

Reply:
Hi @eliben
Sorry for delay in response. I have removed CombineVectors as it is not needed in newer implementation. Previously, batchTexts function was splitting text into parts, then we request for embeddings and combine them to get embeddings for required text. Newer implementation of batchTexts which is inspired from js library doesnt split the text instead it batches the texts and hence we dont need to use CombineVectors.

Assume batchSize = 2 for following example

Previous implementation:
if input to batchTexts = ["str1", "str2", "str3"] it outputs [["st", "r1"], ["st", "r2"], ["st", "r3"]] and then we request for embeddings and we used CombineVectors on embeddings for ["st", "r1"] to get embeddings for "str1"

Newer implementation:
if input to batchTexts = ["str1", "str2", "str3"] it outputs [["str1", "str2"], ["str3"]] and hence we dont require to use CombineVectors after getting embeddings from llm.

Reply:
The issue is that this functionality exists in langchain-py (see for example `_get_len_safe_embeddings`), and is described here:

https://github.com/openai/openai-cookbook/blob/main/examples/Embedding_long_inputs.ipynb

It's not about splitting a list of texts to batches; it's about taking long texts and splitting them to chunks, since embedding models have limits on text lengths.

Reply:
Absolutely, that makes sense. So there's a conflict between how Python and JS have implemented this.

I implemented batchTexts with an inspiration from how its implemented in JS library [embedDocuments](https://github.com/langchain-ai/langchainjs/blob/4e9e23137de8d2bc61dfbcf05e8e35a8f1e89ef5/langchain/src/embeddings/openai.ts#L154) and [batching function](https://github.com/langchain-ai/langchainjs/blob/f465c286193f5193c89bbafee0290070d459f2ec/langchain-core/src/utils/chunk_array.ts#L1)

By batching we generate maximum number of embeddings possible in single request instead of calling create embeddings separately for each text. This is also good from rate limit point of view because different providers impose different rate limits for eg [OpenAI doc](https://platform.openai.com/docs/guides/rate-limits/free-tier-rate-limits). By batching we can req 2048 texts in single req instead of making 2048 reqs (talking about openai only as it  allows to send upto 2048 texts in single req and each text has context limit of 8191 tokens).

Suggestion: We should do both, batching (batch the texts) and chunking (split the individual text) which will make it most efficient.
Looking forward for your views. I will implement chunking+batching if you agree upon it or I can revert changes made by me which will result in only chunking as it was before.